### DISCUSSION TITLE
PstFrom with pre-generated pst-file

### DISCUSSION THREAD

**Discussion by d-strom** [pypest/pyemu#469] [Q&A] [answered]
Hi,

Is it possible to use the PstFrom class to preparere covariance matrices, jacobian matrices and more for a pst control file that has already been set-up?

I have a Preprocessed control file and have been running pest-ies and glm successfully. However I experience lots of Bulls eye behaviour and odd looking parameter fields. Especially with the ies approach.
Looking at the gmdsi tutorials I get the feeling that correctly preparing jcb files and /or prior monte Carlo etc will help me. But they are always a part of the pstFrom process where you prepare the control file from scratch.

Any advice?

Sincerely
David

**Comment by jtwhite79** [comment, marked as answer]
I think the main benefit of pstfrom is that it let's you more easily experiment with how you define parameters and observations (and prior covariances and weights/noise).   If you want to setup some prior cov matrices (or ensembles) the underlying utilities are in pyemu helpers.  But these same functionalities are also available in the pest utilities also...

**Reply by d-strom** [reply]
Are there other means of regularisation associated with pestpp-ies than setting the phimlim condition higher? To get a smoother yet good correlation?

**Comment by jtwhite79** [comment]
In general if you want the posterior parameter realizations to have certain spatial patterns and characteristics, you need to bake those patterns and characteristics into the prior realizations. So you need to generate a prior parameter ensemble with your preferred correlation patterns.  

 In ensemble methods regularization is enforced such that each realization is preferred to stay as much like it prior values as possible.  So you gotta give ies meaningful prior realizations.
